grails-app/controllers/au/org/ala/biocache/hubs/OccurrenceController.groovy

```diff
@@ -725,4 +725,34 @@ class OccurrenceController {
 
         return userPref.expand
     }
+
+    def getAlerts() {
+        String userId = authService?.getUserId()
+        if (userId == null) {
+            response.status = 404
+            render ([error: 'userId must be supplied to get alerts'] as JSON)
+        } else {
+            render webServicesService.getAlerts(userId) as JSON
+        }
+    }
+
+    def addMyAnnotationAlert() {
+        String userId = authService?.getUserId()
+        if (userId == null) {
+            response.status = 404
+            render ([error: 'userId must be supplied to add alert'] as JSON)
+        } else {
+            render webServicesService.addMyAnnotationAlert(userId) as JSON
+        }
+    }
+
+    def deleteMyAnnotationAlert() {
```

Comment:
I'm not sure I understand why this is a `delete` operation. I thought "my annotations" was a permanent alert type that is either on or off. It seems strange to delete it...

Reply:
Hi Nick, yes we can turn it on/off.

On alerts side, it's like a custom query since different user has different query_url (...assertion_user_id=xxx).

So 
- when you turn it on (call `addMyAnnotaionAlert`) a query is created in `alerts`.
- when you turn it off(call `deleteMyAnnotationAlert`) that query in `alerts` is **deleted** (actually query + notification + queryresult all deleted. Delete notification is enough for stopping getting notification though)

Reply:
OK can leave it as is... I'd argue the public interface for the service should just be on/off, as the client doesn't need to know that records are being added and then deleted.... it makes it seem more complex and potentially confusing for clients to implement - "Why is it deleting the annotation? Will the user never be able to turn it on again in the future if its deleted?" I had these questions when I read the code...